### PR TITLE
HPCC-12791 Paste not working in ECL Playground

### PR DESCRIPTION
**Pull request by GordonSmith**
Update to latest CodeMirror
Include new "Find" functionality

Fixes HPCC-12791
Fixes HPCC-11658

Signed-off-by: Gordon Smith gordon.smith@lexisnexis.com

### CODE REVIEW

**Comment by GordonSmith** [comment]
@miguelvazq Please Review
@wangkx Please Review the PackageManager changes.

**Comment by wangkx** [comment]
The ECLPlayground does not work when I run the code of GordonSmith:HPCC-12791. The esp/src/CodeMirror2/lib/codemirror.js and several other files are removed. Do you want to move those files to some other folder? If those files are not needed any more, have you removed the links to those files from stub.htm?

**Comment by GordonSmith** [comment]
Good catch, my changes to stub.htm hadn't made it in...  You can try again...

**Comment by wangkx** [comment]
The changes for PackageManager looks good.

File: esp/src/eclwatch/templates/PackageSourceWidget.html [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <div class="${baseClass}">
     <div id="${id}BorderContainer" class="${baseClass}BorderContainer" style="width: 100%; height: 100%; overflow: hidden" data-dojo-props="splitter: false, gutters:false" data-dojo-type="dijit.layout.BorderContainer">
-        <div id="${id}XMLContent" class="centerPanel" data-dojo-props="region: 'center'" data-dojo-type="dijit.layout.ContentPane">
+        <div id="${id}XMLContent" class="centerPanel" style="padding:0px" data-dojo-props="region: 'center'" data-dojo-type="dijit.layout.ContentPane">
```
Comment (miguelvazq) [inline esp/src/eclwatch/templates/PackageSourceWidget.html]: Terminate the style with a ";" like above.

File: esp/src/eclwatch/templates/ECLSourceWidget.html [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <div class="${baseClass}">
     <div id="${id}BorderContainer" class="${baseClass}BorderContainer" style="width: 100%; height: 100%; overflow: hidden" data-dojo-props="splitter: false, gutters:false" data-dojo-type="dijit.layout.BorderContainer">
-        <div id="${id}EclContent" class="centerPanel" data-dojo-props="region: 'center'" data-dojo-type="dijit.layout.ContentPane">
+        <div id="${id}EclContent" class="centerPanel" style="padding:0px" data-dojo-props="region: 'center'" data-dojo-type="dijit.layout.ContentPane">
```
Comment (miguelvazq) [inline esp/src/eclwatch/templates/ECLSourceWidget.html]: Terminate the style with a ";" like above.

**Comment by miguelvazq** [comment]
All else looks fine.

File: esp/src/eclwatch/templates/PackageSourceWidget.html (hunk 1 above)
Comment (GordonSmith) [inline esp/src/eclwatch/templates/PackageSourceWidget.html]: ; is a statement separator and not a statement terminator - which is why this is ok.

File: esp/src/eclwatch/templates/ECLSourceWidget.html (hunk 2 above)
Comment (GordonSmith) [inline esp/src/eclwatch/templates/ECLSourceWidget.html]: ; is a statement separator and not a statement terminator - which is why this is ok.

**Pull request merged**